### PR TITLE
YLP-1346 show basal data in yourloops

### PR DESCRIPTION
**Pull request by Nas84**

### CODE REVIEW

File: data/api_v1_data_helper.go [hunk 1]
```diff
@@ -33,8 +33,8 @@ func getDataV1Params(res *httpResponseWriter) (*apiDataParams, *detailedError) {
 	startDate := query.Get("startDate")
 	endDate := query.Get("endDate")
 	withPumpSettings := query.Get("withPumpSettings") == "true"
-	basalBucket := query.Get("basalBucket") == "true"
-	cbgBucket := (query.Get("cbgBucket") == "true" || query.Get("cbgBucket") == "")
+	basalBucket := query.Get("basalBucket") == "true" || query.Get("basalBucket") == ""
+	cbgBucket := query.Get("cbgBucket") == "true" || query.Get("cbgBucket") == ""
```
Comment (CedricPerotto) [inline data/api_v1_data_helper.go:37]: As all cbg are now in data-read db (and are missing in legacy data.deviceData), we may set this cbgBucket variable to true (or get rid of it).

Comment (Nas84) [reply data/api_v1_data_helper.go]: done

File: data/api_v1_data_helper.go [hunk 2]
```diff
@@ -257,7 +256,7 @@ func writeCbgs(ctx context.Context, p *writeFromIter) error {
 		for i, sample := range bucket.Samples {
 			datum := make(map[string]interface{})
 			// Building a fake id (bucket.Id/range index)
-			datum["id"] = fmt.Sprintf("%s_%d", bucket.Id, i)
+			datum["id"] = fmt.Sprintf("%s_%s_%d", "cbg", bucket.Id, i)
```
Comment (CedricPerotto) [inline data/api_v1_data_helper.go:259]:
```suggestion
			datum["id"] = fmt.Sprintf("cbg_%s_%d", bucket.Id, i)
```

File: data/api_v1_data_helper.go [hunk 3]
```diff
@@ -295,7 +294,7 @@ func writeBasals(ctx context.Context, p *writeFromIter) error {
 		for i, sample := range bucket.Samples {
 			datum := make(map[string]interface{})
 			// Building a fake id (bucket.Id/range index)
-			datum["id"] = fmt.Sprintf("%s_%d", bucket.Id, i)
+			datum["id"] = fmt.Sprintf("%s_%s_%d", "basal", bucket.Id, i)
```
Comment (CedricPerotto) [inline data/api_v1_data_helper.go:297]:
```suggestion
			datum["id"] = fmt.Sprintf("basal_%s_%d", bucket.Id, i)
```

File: data/api.go [hunk 4]
```diff
@@ -59,9 +59,11 @@ var (
 	errorTideV2Http        = detailedError{Status: http.StatusInternalServerError, Code: "tidev2_error", Message: "internal server error"}
 	errorInvalidParameters = detailedError{Status: http.StatusBadRequest, Code: "invalid_parameters", Message: "one or more parameters are invalid"}
 	errorNotfound          = detailedError{Status: http.StatusNotFound, Code: "data_not_found", Message: "no data for specified user"}
+	readBasalBucket        = false
 )
 
-func InitAPI(storage store.Storage, auth auth.ClientInterface, permsClient opa.Client, schemaV store.SchemaVersion, logger *log.Logger, V2Client tideV2Client.ClientInterface) *API {
+func InitAPI(storage store.Storage, auth auth.ClientInterface, permsClient opa.Client, schemaV store.SchemaVersion, logger *log.Logger, V2Client tideV2Client.ClientInterface, envReadBasalBucket bool) *API {
+	readBasalBucket = envReadBasalBucket
```
Comment (CedricPerotto) [inline data/api.go:66]: why setting it as a global and not in the API struct ? 

My point is that when reading the code (outside of the InitAPI func), it's not very clear to know from where it comes from.

**Review by CedricPerotto — CHANGES_REQUESTED** [review]
Looks like it works (need some small changes in k8s template)... 

Just a few my minor requests

File: data/api.go (hunk 4 above)
Comment (Nas84) [reply data/api.go]: no particular reason, will change

File: data/api_v1_data_helper.go [hunk 5]
```diff
@@ -33,7 +33,7 @@ func getDataV1Params(res *httpResponseWriter) (*apiDataParams, *detailedError) {
 	startDate := query.Get("startDate")
 	endDate := query.Get("endDate")
 	withPumpSettings := query.Get("withPumpSettings") == "true"
-	basalBucket := query.Get("basalBucket") == "true" || query.Get("basalBucket") == ""
+	basalBucket := readBasalBucket
```
[the remark this answers is not in the document]
Comment (Nas84) [reply data/api_v1_data_helper.go]: yes 👍🏽

File: data/api.go [hunk 6]
```diff
@@ -61,14 +62,16 @@ var (
 	errorNotfound          = detailedError{Status: http.StatusNotFound, Code: "data_not_found", Message: "no data for specified user"}
 )
 
-func InitAPI(storage store.Storage, auth auth.ClientInterface, permsClient opa.Client, schemaV store.SchemaVersion, logger *log.Logger, V2Client tideV2Client.ClientInterface) *API {
+func InitAPI(storage store.Storage, auth auth.ClientInterface, permsClient opa.Client, schemaV store.SchemaVersion, logger *log.Logger, V2Client tideV2Client.ClientInterface, envReadBasalBucket bool) *API {
+	//readBasalBucket = envReadBasalBucket
```
Comment (CedricPerotto) [inline data/api.go:66]:
```suggestion
```

**Review by CedricPerotto — CHANGES_REQUESTED** [review]

**Review by CedricPerotto — APPROVED** [review]

**Pull request merged**